data/org.freedesktop.portal.GlobalShortcuts.xml

```diff
@@ -0,0 +1,134 @@
+<?xml version="1.0"?>
+<!--
+ Copyright (C) 2022 Aleix Pol Gonzalez <aleixpol@kde.org>
+
+ This library is free software; you can redistribute it and/or
+ modify it under the terms of the GNU Lesser General Public
+ License as published by the Free Software Foundation; either
+ version 2 of the License, or (at your option) any later version.
+
+ This library is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ Lesser General Public License for more details.
+
+ You should have received a copy of the GNU Lesser General Public
+ License along with this library. If not, see <http://www.gnu.org/licenses/>.
+
+ Author: Aleix Pol Gonzalez <aleixpol@kde.org>
+-->
+
+<node name="/" xmlns:doc="http://www.freedesktop.org/dbus/1.0/doc.dtd">
+  <!--
+      org.freedesktop.portal.GlobalShortcut:
+      @short_description: GlobalShortcut portal backend interface
+
+      This this portal lets applications register global shortcuts so they can
+      act regardless of the system state upon an input event.
+
+      To use this, an app will first create a session under which its actions
+      will be collected. Then shortcuts can be registered, listed and eventually
+      unregistered. It's up to the portal implementation to perform the registration
+      and tracking of that shortcut throughout the session's lifespan.
```

Comment:
fwiw, I found this description to be confusing because afaict you cannot register [1] a shortcut here, you can merely *bind* [2] to existing shortcuts. So aiui you can set up a "zoom in" shortcut in the compositor which will then show up in the list below. The application can "bind" to that shortcut but only because it exists.

An application cannot say "here's a new shortcut 'rotate left'" with this interface.

[1] "register" being interpreted as "create a new thing and add it to a lookup table"
[2] "bind" being interpreted as "make known that I want to use this thing"

Reply:
I've fixed this description, it was a left-over from the discussion that happened in the originating ticket. Please tell me if it makes sense to you now.

Reply:
> If so, don't you need a description to go from application to portal too? Something like Bind("zoom-in", "Zoom into the canvas") so the portal can use a human-readable version to display what the user should create a mapping for

When we CreateSession we already pass a text explaining what the shortcut is for. The portal can use that.

> And if you want to future-proof this, make it an a{sv} passed in via options, so you can later add things like enabled.

We can always have more methods to interact with the shortcut. We could have a setEnabled(shortcut-id) kind of call.

> I wonder how that'll work with translations btw but i18n is not my forte. Or switching keyboard layouts after the shortcut was bound, because "Press Ctrl+A" might as well now be an effective "Ctrl+Q". This part is largely compositor-dependent though.

The portal will offer strings in the user's language. And yes, since it's for the user it can be either "Ctrl+A", "Control+A", "Press Control and without letting go, press A" and any variation. Obviously, conciseness is advisable. FWIW, other more esoteric options would be possible like "Say Banana to your toaster", again responsibility of your portals implementation.

Reply:
> When we CreateSession we already pass a text explaining what the shortcut is for. The portal can use that.

indeed, I read over that. for the documentation, maybe use different terms like "description" for what the shortcut is supposed to do and "definition" for what the shortcut is set to trigger by? or whatever else the thesaurus suggests. Makes things easier for denser people like me :)